src/pip/_internal/commands/cache.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import absolute_import
+
+import logging
+import os
+import textwrap
+
+import pip._internal.utils.filesystem as filesystem
+from pip._internal.cli.base_command import Command
+from pip._internal.cli.status_codes import ERROR, SUCCESS
+from pip._internal.exceptions import CommandError, PipError
+from pip._internal.utils.typing import MYPY_CHECK_RUNNING
+
+if MYPY_CHECK_RUNNING:
+    from optparse import Values
+    from typing import Any, List
+
+
+logger = logging.getLogger(__name__)
+
+
+class CacheCommand(Command):
+    """
+        Inspect and manage pip's wheel cache.
+
+        Subcommands:
+
+        info:
+            Show information about the cache.
+        list:
+            List filenames of packages stored in the cache.
+        remove:
+            Remove one or more package from the cache.
+        purge:
+            Remove all items from the cache.
+
+        <pattern> can be a glob expression or a package name.
+    """
+
+    usage = """
+        %prog info
+        %prog list [<pattern>]
+        %prog remove <pattern>
+        %prog purge
+    """
+
+    def __init__(self, *args, **kw):
+        # type: (*Any, **Any) -> None
+        super(CacheCommand, self).__init__(*args, **kw)
+
+    def run(self, options, args):
+        # type: (Values, List[Any]) -> int
+        handlers = {
+            "info": self.get_cache_info,
+            "list": self.list_cache_items,
+            "remove": self.remove_cache_items,
+            "purge": self.purge_cache,
+        }
+
+        # Determine action
+        if not args or args[0] not in handlers:
+            logger.error("Need an action ({}) to perform.".format(
+                ", ".join(sorted(handlers)))
+            )
+            return ERROR
+
+        action = args[0]
+
+        # Error handling happens here, not in the action-handlers.
+        try:
+            handlers[action](options, args[1:])
+        except PipError as e:
+            logger.error(e.args[0])
+            return ERROR
+
+        return SUCCESS
+
+    def get_cache_info(self, options, args):
+        # type: (Values, List[Any]) -> None
+        num_packages = len(self._find_wheels(options, '*'))
+
+        cache_location = self._wheels_cache_dir(options)
+        cache_size = filesystem.friendly_directory_size(cache_location)
+
+        message = textwrap.dedent("""
+            Cache info:
+              Location: {location}
+              Packages: {package_count}
+              Size:     {size}
```

Comment:
```suggestion
            Location: {location}
            Size: {size}
            Number of wheels: {package_count}
```

Reply:
Which of these do you think is better or more in-line with how pip typically does things?:

```
            Location: {location}
            Size: {size}
            Number of wheels: {package_count}
```

or:

```
            Location:         {location}
            Size:             {size}
            Number of wheels: {package_count}
```

Reply:
The first matches the behaviour of `pip show PKG`, and with the suggested change from "Packages:" to "Number of wheels:" I think the amount of whitespace is a little jarring. So I'd prefer the non-aligned version. I'm equally OK with your original version, though, TBH (in that case, aligning is OK because the lengths of the various headings are similar).

Ultimately, I think either is fine - this is for human consumption, not machine readability, so the key is readability which is always subjective.

Reply:
I think I'll go with what @pradyunsg suggested because I also don't have a particularly strong opinion on it and it's consistent with other commands. There's discussion in another issue about adding a `--json` flag for getting a machine-readable version of `pip cache`, too, which would give us the freedom to tweak the human-readable format freely later on.